tests/linearizability/model.go

```diff
@@ -105,92 +105,99 @@ var etcdModel = porcupine.Model{
-	if len(state.PossibleValues) == 0 {
-		state.Key = request.Key
-		if ok, v := initValueRevision(request, response); ok {
-			state.PossibleValues = append(state.PossibleValues, v)
+func step(states []EtcdState, request EtcdRequest, response EtcdResponse) (bool, []EtcdState) {
+	if len(states) == 0 {
+		if ok, v := initState(request, response); ok {
+			states = append(states, v)
 		}
-		return true, state
-	}
-	if state.Key != request.Key {
-		panic("multiple keys not supported")
+		return true, states
 	}
 	if response.Err != nil {
-		for _, v := range state.PossibleValues {
-			newV, _ := stepValue(v, request)
-			state.PossibleValues = append(state.PossibleValues, newV)
+		for _, s := range states {
+			newState, _ := stepState(s, request)
+			states = append(states, newState)
 		}
 	} else {
 		var i = 0
-		for _, v := range state.PossibleValues {
-			newV, expectedResponse := stepValue(v, request)
+		for _, s := range states {
+			newState, expectedResponse := stepState(s, request)
 			if expectedResponse == response {
-				state.PossibleValues[i] = newV
+				states[i] = newState
 				i++
 			}
 		}
-		state.PossibleValues = state.PossibleValues[:i]
+		states = states[:i]
 	}
-	return len(state.PossibleValues) > 0, state
+	return len(states) > 0, states
 }
 
-func initValueRevision(request EtcdRequest, response EtcdResponse) (ok bool, v ValueRevision) {
+func initState(request EtcdRequest, response EtcdResponse) (ok bool, s EtcdState) {
 	if response.Err != nil {
-		return false, ValueRevision{}
+		return false, EtcdState{}
 	}
+	kvs := map[string]string{}
 	switch request.Op {
 	case Get:
-		return true, ValueRevision{
-			Value:    response.GetData,
-			Revision: response.Revision,
+		if response.GetData != "" {
+			kvs[request.Key] = response.GetData
+		}
+		return true, EtcdState{
+			Revision:  response.Revision,
+			KeyValues: kvs,
 		}
 	case Put:
-		return true, ValueRevision{
-			Value:    request.PutData,
-			Revision: response.Revision,
+		kvs[request.Key] = request.PutData
+		return true, EtcdState{
+			Revision:  response.Revision,
+			KeyValues: kvs,
 		}
 	case Delete:
-		return true, ValueRevision{
-			Value:    "",
-			Revision: response.Revision,
+		return true, EtcdState{
+			Revision:  response.Revision,
+			KeyValues: kvs,
 		}
 	case Txn:
 		if response.TxnSucceeded {
-			return true, ValueRevision{
-				Value:    request.TxnNewData,
-				Revision: response.Revision,
+			kvs[request.Key] = request.TxnNewData
+			return true, EtcdState{
+				Revision:  response.Revision,
+				KeyValues: kvs,
 			}
 		}
-		return false, ValueRevision{}
+		return false, EtcdState{}
 	default:
 		panic("Unknown operation")
 	}
 }
 
-func stepValue(v ValueRevision, request EtcdRequest) (ValueRevision, EtcdResponse) {
+func stepState(s EtcdState, request EtcdRequest) (EtcdState, EtcdResponse) {
+	newKVs := map[string]string{}
+	for k, v := range s.KeyValues {
+		newKVs[k] = v
+	}
```

Comment:
Please consider to add a clone method something like below,

```
func (s EtcdState) clone() EtcdState {
	newKVs := map[string]string{}
	for k, v := range s.KeyValues {
		newKVs[k] = v
	}  
	return EtcdState{
		Revision: s.Revision,
		KeyValues: newKVs,
	}
}

```

Reply:
This comment isn't valid any more given https://github.com/etcd-io/etcd/pull/14924/files#r1045433830